README.md

```diff
@@ -5,34 +5,32 @@
-can be wrapped with integration stability patterns without code changes. 
-Besides, when the chain becomes more complex and additional failure scenarios are known, 
-failure logic can be adjusted with fallback configuration (no changes in the business logic required).
+Knot.x Fragments is designed to build fault-tolerant, reactive, back-end integrations such as:
+- [API Gateway](https://github.com/Knotx/knotx-example-project#api-gateway--web-api)
+- Backend For Frontend (BFF) for single-page applications (SPA)
+- [Web API](https://github.com/Knotx/knotx-example-project#api-gateway--web-api) (both REST and GraphQL)
 
-Knot.x Fragments is designed to build fault-tolerant, back-end integrations such as:
-- API Gateways, Backend For Frontend (BFF) for single-page applications (SPA), Web APIs
-- documents processing (HTML, JSON, PDF etc) with a templating engine support
+Its initial purpose was a [templating solution](https://github.com/Knotx/knotx-example-project#template-processing) 
+combined dynamic data (from external data sources, 3rd party API etc.) with static content (HTML, 
+JSON, PDF etc) that comes from various content stores (such as Wordpress, Drupal, Magnolia or 
+Adobe Experience Manager).
 
 ## Modules
 
+- [Fragments API](https://github.com/Knotx/knotx-fragments/tree/master/api) - defines a [**Fragment**](https://github.com/Knotx/knotx-fragments/tree/master/api#fragment) 
+and a [Fragment Operation](https://github.com/Knotx/knotx-fragments/tree/master/api#fragment-operation)
+- [Actions](https://github.com/Knotx/knotx-fragments/tree/master/action) - extendable 
+library of fragment operations that simplifies integration with APIs and provides such stability 
+patterns as a circuit breaker mechanism
 - [Fragments Supplier](https://github.com/Knotx/knotx-fragments/tree/master/supplier) - converts an Http request into one or more [**Fragments**](https://github.com/Knotx/knotx-fragments/tree/master/api#knotx-fragment-api)
-    - [HTML Splitter](https://github.com/Knotx/knotx-fragments/tree/master/supplier/html-splitter)
-    - [Single Fragment Supplier](https://github.com/Knotx/knotx-fragments/tree/master/supplier/single-fragment)
-- [Fragments Handler](https://github.com/Knotx/knotx-fragments/tree/master/handler) - configures tasks for fragments and delegates processing to Engine 
-    - [Fragment Execution Log Consumer](https://github.com/Knotx/knotx-fragments/tree/master/handler/consumer) - exposes data from evaluated fragments
-- [Fragments Engine](https://github.com/Knotx/knotx-fragments/tree/master/engine) - evaluates fragments (tasks) with map-reduce nature
 - [Fragments Assembler](https://github.com/Knotx/knotx-fragments/tree/master/assembler) - merges Fragments into a single response
-
-Each module contains its own documentation inside.
+- [Task](https://github.com/Knotx/knotx-fragments/tree/master/task) - processes fragments and applies a configured business logic assigned to them
```

Comment:
```suggestion
- [Task](https://github.com/Knotx/knotx-fragments/tree/master/task) - processes Fragments and applies a configured business logic assigned to them
```

Reply:
We use capital letter only first time, when we define the item. Later we use: `fagment`, `action`.